tools/flaky-test-retryer/github_commenter.go

```diff
@@ -105,7 +158,14 @@ func (gc *GithubClient) getOldComment(org, repo string, pull int) (*github.Issue
 	}
 	var match *github.IssueComment
 	for _, comment := range comments {
-		found, err := regexp.Match(identifier, []byte(comment.GetBody()))
+		body := []byte(comment.GetBody())
+		found, err := regexp.Match(legacyIdentifier, body)
+		if err != nil || !found {
+			if testNameFromComment := regexp.MustCompile(reTestIdentifier).FindStringSubmatch(comment.GetBody()); len(testNameFromComment) >= 2 {
```

Comment:
why do we wnat to do this check even when err != nil ?

Reply:
good catch, that was the part I didn't touch. Changed to make it more reasonable